sound/soc/sof/ipc.c

```diff
@@ -400,14 +400,19 @@ static void ipc_msgs_rx(struct work_struct *work)
 			if (sof_ops(sdev)->fw_ready)
 				err = sof_ops(sdev)->fw_ready(sdev, cmd);
 			if (err < 0) {
-				dev_err(sdev->dev, "error: DSP firmware boot timeout %d\n",
+				dev_err(sdev->dev,
+					"error: fw ready IPC failed with error %d\n",
 					err);
+
+				/* fw_ready() error indicates ABI mismatch */
```

Comment:
we'd need to make that explicit in the description of the callback and implementations.

Reply:
@plbossart fixed now